03/app.js

```diff
@@ -10,18 +10,43 @@ class App extends React.Component {
     state = {
         cart: [],
     }
-    
+
     render() {
         return (
             <section>
-                <Category />
-                <Cart />
+                <Category data={data} cart={this.state.cart} onClick={this.addToCart} />
+                <Cart cart={this.state.cart} onClick={this.removeFromCart} />
             </section>
         )
     }
+
+    addToCart = (product) => {
+        this.setState( state => {
+            return {
+                cart: [...state.cart, product]
+            }
+        })
+    }
+
+    removeFromCart = (product) => {
+        this.setState( state => {
+            return {
+                cart: [...state.cart.filter(
+                    item => item !== product
```

Comment:
Często lepiej jest porównywać `id` zamiast całych obiektów (pamiętając, że musi to być referencja, aby zadziałało)